examples/advanced_projects/efficient_bounce_back_obstacle/linear_interpolated_bounce_back_boundary.py

```diff
@@ -0,0 +1,217 @@
+                self.f_collided_lt[:, 0] = torch.clone(f_collided[self.f_index_lt[:, 0],  # q
+                                                              self.f_index_lt[:, 1],  # x
+                                                              self.f_index_lt[:, 2]])  # y
+                self.f_collided_lt[:, 1] = torch.clone(f_collided[self.opposite_tensor[self.f_index_lt[:,0]],  # q
+                                                              self.f_index_lt[:, 1],  # x
+                                                              self.f_index_lt[:, 2]])  # y
+            if len(self.f_collided_gt) != 0:
+                self.f_collided_gt[:, 0] = torch.clone(f_collided[self.f_index_gt[:, 0],  # q
+                                                              self.f_index_gt[:, 1],  # x
+                                                              self.f_index_gt[:, 2]])  # y
+                self.f_collided_gt[:, 1] = torch.clone(f_collided[self.opposite_tensor[self.f_index_gt[:,0]],  # q
+                                                              self.f_index_gt[:, 1],  # x
+                                                              self.f_index_gt[:, 2]])  # y
+        if self.flow.stencil.d == 3:
+            if len(self.f_collided_lt) != 0:
+                self.f_collided_lt[:, 0] = torch.clone(f_collided[self.f_index_lt[:, 0],  # q
+                                                              self.f_index_lt[:, 1],  # x
+                                                              self.f_index_lt[:, 2],  # y
+                                                              self.f_index_lt[:, 3]])  # z
+                self.f_collided_lt[:, 1] = torch.clone(f_collided[self.opposite_tensor[self.f_index_lt[:,0]],  # q
+                                                              self.f_index_lt[:, 1],  # x
+                                                              self.f_index_lt[:, 2],  # y
+                                                              self.f_index_lt[:, 3]])  # z
+            if len(self.f_collided_gt) != 0:
+                self.f_collided_gt[:, 0] = torch.clone(f_collided[self.f_index_gt[:, 0],  # q
+                                                                  self.f_index_gt[:, 1],  # x
+                                                                  self.f_index_gt[:, 2],  # y
+                                                                  self.f_index_gt[:, 3]])  # z
+                self.f_collided_gt[:, 1] = torch.clone(f_collided[self.opposite_tensor[self.f_index_gt[:, 0]],  # q
+                                                                  self.f_index_gt[:, 1],  # x
+                                                                  self.f_index_gt[:, 2],  # y
+                                                                  self.f_index_gt[:, 3]])  # z
+    #<<< OLD version "semi hardcoded"
+
+    # TODO: find a way to use pre- and post-Streaming Populations for bounce...
```

Comment:
if this still needs to be done, can you create a separate issue? or is this still to-do in this PR?